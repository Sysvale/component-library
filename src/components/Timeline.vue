<template>
	<span id="timeline">
		<div
			class="panel-body-content"
		>
			<div
				v-for="(event, index) in history"
				:key="index"
			>
				<div class="event-row">
					<div>
						<p
							class="regular fs-14 text-secondary date-text"
						>
							{{ event.date }}
						</p>
					</div>
					<div
						class="timeline-container"
					>
						<span
							class="event-pin"
						/>
						<div
							v-if="((index + 1) < history.length) && (history.length > 1)"
							class="timeline"
						/>
					</div>
					<div class="event-info-container">
						<div>
							<p
								class="mb-0 event-title regular">
								{{ event.title }}
							</p>
						</div>
						<div class="text-muted fs-14 mt-1">
							{{ event.text }}
						</div>
						<hr
							class="event-row-divider"
							v-if="(index + 1) < history.length"
						>
					</div>
				</div>
			</div>
		</div>
	</span>
</template>

<script>
export default {
	props: {
		history: {
			type: Array,
			default: () => [],
			description: 'The array that specifies the events that willbe displayed in the timeline.',
			required: true,
		},
	},
};
</script>
<style>
#timeline .panel-body-content {
	padding: 24px !important;;
	overflow: auto;
	max-height: 350px;
	width: 100% !important;;
	scrollbar-color: #CDCDCD #FFFFFF;
	scrollbar-width: thin;
}

#timeline .event-title {
	color: #182532;
	font-weight: 600;
}

#timeline .event-pin {
	height: 12px !important;;
	width: 11.5px !important;;
	border-radius: 50% !important;;
	background-color: #2EB88D;
	border: 1px solid #2EB88D;
}

#timeline .timeline {
	height: 100% !important;
	width: 1px !important;;
	background-color: #CDCDCD;
}

#timeline .timeline-container {
	display: flex;
	flex-direction: column;
	align-items: center;
}

#timeline .event-row {
	display: flex;
	justify-content: space-between;
}

#timeline .event-info-container {
	width: 70% !important;;
	margin-top: -8px;
}

#timeline .panel-body-content::-webkit-scrollbar {
	width: 0.35rem;
}

#timeline .panel-body-content::-webkit-scrollbar-thumb {
	border-radius: 4px;
	background: #CDCDCD#FFFFFF
}

#timeline .date-text {
	text-transform: Capitalize;
	align-self: flex-end;
	margin-top: -4px;
	margin-bottom: 0px;
}

#timeline .text-muted {
	color: #697580;
}
</style>
