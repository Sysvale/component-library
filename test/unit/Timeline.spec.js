import { mount, createLocalVue } from '@vue/test-utils';
import BootstrapVue from 'bootstrap-vue';
import Timeline from '../../src/components/Timeline.vue';

const localVue = createLocalVue();
localVue.use(BootstrapVue);

const history = [
	{
		date: '22/09/2020',
		title: 'Bloqueio nível 1',
		text: 'Usuário: Uncle Bob',
	},
	{
		date: '22/09/2020',
		title: 'Bloqueio nível 1',
		text: 'Usuário: Uncle Bob',
	},
	{
		date: '22/09/2020',
		title: 'Bloqueio nível 1',
		text: 'Usuário: Uncle Bob',
	},
];

test('Component is mounted properly', () => {
	const wrapper = mount(Timeline, {
		localVue,
		propsData: {
			history: history,
		},
	});

	expect(wrapper).toMatchSnapshot();
});

test('If the number of rows are the same of the length of the array', () => {
	const wrapper = mount(Timeline, {
		localVue,
		propsData: {
			history: history,
		},
	});
	
	expect(wrapper.findAll('.event-row').length).toBe(history.length);
});

test('If the UI elements of the timeline are displayed properly', () => {
	const wrapper = mount(Timeline, {
		localVue,
		propsData: {
			history: history,
		},
	});

	expect(wrapper.findAll('.event-pin').length).toBe(history.length);
	expect(wrapper.findAll('.timeline').length).toBe(history.length - 1);
	expect(wrapper.findAll('.event-row-divider').length).toBe(history.length - 1);
});
