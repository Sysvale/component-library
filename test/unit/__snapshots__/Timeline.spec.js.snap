// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component is mounted properly 1`] = `
<span id="timeline"><div class="panel-body-content"><div><div class="event-row"><div><p class="regular fs-14 text-secondary date-text">
						22/09/2020
					</p></div> <div class="timeline-container"><span class="event-pin"></span>
<div class="timeline"></div>
</div>
<div class="event-info-container">
  <div>
    <p class="mb-0 event-title regular">
      Bloqueio nível 1
    </p>
  </div>
  <div class="text-muted fs-14 mt-1">
    Usuário: Uncle Bob
  </div>
  <hr class="event-row-divider">
</div>
</div>
</div>
<div>
  <div class="event-row">
    <div>
      <p class="regular fs-14 text-secondary date-text">
        22/09/2020
      </p>
    </div>
    <div class="timeline-container"><span class="event-pin"></span>
      <div class="timeline"></div>
    </div>
    <div class="event-info-container">
      <div>
        <p class="mb-0 event-title regular">
          Bloqueio nível 1
        </p>
      </div>
      <div class="text-muted fs-14 mt-1">
        Usuário: Uncle Bob
      </div>
      <hr class="event-row-divider">
    </div>
  </div>
</div>
<div>
  <div class="event-row">
    <div>
      <p class="regular fs-14 text-secondary date-text">
        22/09/2020
      </p>
    </div>
    <div class="timeline-container"><span class="event-pin"></span>
      <!---->
    </div>
    <div class="event-info-container">
      <div>
        <p class="mb-0 event-title regular">
          Bloqueio nível 1
        </p>
      </div>
      <div class="text-muted fs-14 mt-1">
        Usuário: Uncle Bob
      </div>
      <!---->
    </div>
  </div>
</div>
</div>
</span>
`;
